addons/mail/static/src/core/common/message_model.js

```diff
@@ -346,12 +372,16 @@ export class Message extends Record {
             mentionedChannels,
             mentionedPartners,
         });
+        let prettyBody = await prettifyMessageContent(body, validMentions);
+        if (this.thread.model === "discuss.channel") {
+            prettyBody = `<odoo-markdown>${prettyBody}</odoo-markdown>`;
+        }
```

Comment:
This is probably not a good way of doing it.

There should be no processing of any input at all inside markdown code blocks in particular.